include/mgba-util/platform/posix/threading.h

```diff
@@ -29,6 +29,18 @@ static inline int MutexInit(Mutex* mutex) {
 	return pthread_mutex_init(mutex, 0);
 }
 
+static inline int MutexInitRecursive(Mutex* mutex) {
```

Comment:
Why did you add this? You'd need to port it to all of the threading platforms and I don't think it's necessary.

Reply:
In your callback code in `MultilplayerController` you have another lock. I noticed it is unnecessary if mutex is used with lockstep anw. So make it recursive. Anw, the lock call in callback might not be needed. So the recursive mutex is not really neccessary.
For the platform porting, right now i only see win32 and pthread ans they are already ported